lib/async_experiments.rb

```diff
@@ -12,20 +14,18 @@ def self.statsd=(statsd)
   end
 
   def self.get_experiment_data(experiment_name)
-    mismatched_responses = Sidekiq.redis { |redis|
-      redis.lrange("experiments:#{experiment_name}:mismatches", 0, -1)
-    }
-
-    mismatched_responses.map { |json|
-      parsed = JSON.parse(json)
+    mismatched_responses = Sidekiq.redis do |redis|
+      mismatch_enumerator = redis.scan_each(
+        match: "experiments:#{experiment_name}:mismatches:*",
+      )
+      retrieve = -> (key) { redis.get(key) }
+      mismatch_enumerator.map(&retrieve).compact.map { |json| JSON.parse(json) }
+    end
 
-      missing, other = parsed.partition {|(operator, _, _)|
-        operator == "-"
-      }
+    mismatched_responses.map do |parsed|
+      missing, other = parsed.partition { |(operator)| operator == "-" }
```

Comment:
They're used to access the first item in the array, it might be that there is a more idiomatic way to represent this I'm not sure :-/